File: frontend/training-platform-ui/src/app/features/trainer/pages/models/Lesson-Progress-Response.ts
export interface LessonProgressResponse {
  id: number;

  learnerId: number;

  lessonId: number;

  lessonTitle: string;

  courseId: number;

  completed: boolean;

  completedAt: string | null;

  createdAt: string;

  updatedAt: string;
}

File: backend/training-platform/src/main/java/com/trainingplatform/progress/dto/response/LessonProgressResponse.java
package com.trainingplatform.progress.dto.response;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
public class LessonProgressResponse {

    private Long id;

    private Long learnerId;

    private Long lessonId;

    private String lessonTitle;

    private Long courseId;

    private Boolean completed;

    private LocalDateTime completedAt;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}